content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,662 @@
+   }
+   ```
+5. Open the **Editor** and you should see your tree swaying with the leaves still bending. If you don't, try opening the **Material Editor** and increasing the wind parameters.
+
+Amazing, our tree now sways and reacts to wind! Try to place multiple trees and observe how the trees sway differently when close together versus farther away. Also, add some lighting to make the trees pop!
+
+{{< video src="/images/atom-guide/vegetation-bending-tutorial/tree3.mp4" autoplay="true" loop="true" width="100%" muted="true" info="Video of trees swaying." >}}
+
+### Add motion vectors
+Because our tree is moving, we can add a *motion vector pass* so we can choose to add motion blur or other effects. For example, we have a partially implemented [Temporal Anti-Aliasing (TAA)](../../../atom-guide/features/taa.md) feature which uses the motion vectors. 
+
+If you take a look at the `MeshMotionVectorVegetationBending.azsl` file, you can see that the vertex shader looks similar to the other vertex shaders we have seen, but there's a new output `OUT.m_worldPosPrev`. This is the position of the vector in the last time frame. The pixel shader uses both the previous vector position and the current one to calculate the motion vector. 
+
+However, we don't have an vertex shader input that gives us the previous position. Therefore, in the vertex shader, we want to calculate the bending for not only the current time as we have been, but also for the previous time frame.
+
+1. Download the two files in the `MeshMotionVectorVegetationBending` folder and move it to `{your-project-path}\Materials\Types`.
+2. The `MeshMotionVectorVegetationBending` files already contain all the previous steps that we have done, like adding the shader option and calling the bending functions. 
+3. Open `VegetationBending.materialtype`. At the bottom of the `shaders` list, add the motion vector pass:
+   
+   ```
+   {
+      "file": "./MeshMotionVectorVegetationBending.shader",
+      "tag": "MeshMotionVector"
+   }
+   ```
+
+Great, now we have included the motion vector shader, so we have to go in and edit the vertex shader.
+
+1. Open `MeshMotionVectorVegetationBending.azsl`.
+2. We want to perform bending on the vertex at the `IN.m_position` but at the previous frame time, with the previous world position. We can use `SceneSrg::m_prevTime` to get the previous frame time. Call our `ProcessBending` function again, but with the appropriate time and world position:
+   ```
+   float currentTime = SceneSrg::m_time;
+   worldPosition = ProcessBending(currentTime, IN.m_position, IN.m_normal, IN.m_optional_color, worldPosition, objectToWorld);
+   float prevTime = SceneSrg::m_prevTime;
+   prevWorldPosition = ProcessBending(prevTime, IN.m_position, IN.m_normal, IN.m_optional_color, prevWorldPosition, prevObjectToWorld);
+   ```
+3. Take a look at the pixel shader to see how the motion vector is calculated! There is no need to edit the pixel shader.
```

Comment:
Take a look at the pixel shader -> Take a look at the vertex shader.

Reply:
Not sure what you meant, do you mean to replace "Take a look at the pixel shader to see how the motion vector is calculated!" with "Take a look at the vertex shader."?

Reply:
Leave it as is. I miss-understood...It makes sense this way in the context of having the whole complete file. Step 2 says edit the vertex shader, step 3 says Look at (but don't modify) the pixel shader.

Originally I was reading the first sentence of step 3 as if it were saying "look at the shader you just modified in step 2 to see how the motion vector is calculated", and thinking that there was something else going on in the vertex shader to calculate the motion vector after this snippit. But I can see now that there is nothing else afterwards in the vertex shader, and you really did mean to look at the pixel shader.

Reply:
Got it!